build/includes/google-cloud.mk

```diff
@@ -35,6 +35,36 @@ gcloud-test-cluster: $(ensure-build-image)
 	$(MAKE) gcloud-auth-cluster
 	$(MAKE) setup-test-cluster
 
+terraform-init:
+	docker run --rm -it $(common_mounts) $(DOCKER_RUN_ARGS) $(build_tag) bash -c '\
+	cd $(mount_path)/build && terraform init && gcloud auth application-default login'
+
+terraform-clean:
+	rm -r ./gke-test-cluster/.terraform
+	rm ./gke-test-cluster/terraform.tfstate*
+
+
+gcloud-terraform-cluster: GCP_PROJECT ?= agones
+gcloud-terraform-cluster: GCP_CLUSTER_LEGACYABAC ?= false
+gcloud-terraform-cluster: GCP_CLUSTER_NODEPOOL_INITIALNODECOUNT ?= 4
+gcloud-terraform-cluster: GCP_CLUSTER_NODEPOOL_MACHINETYPE ?= n1-standard-4
+gcloud-terraform-cluster:
+	test -n "$(GCP_PASSWORD)"  # $$GCP_PASSWORD is undefined
+	docker run --rm -it $(common_mounts) $(DOCKER_RUN_ARGS) $(build_tag) bash -c ' \
+	cd $(mount_path)/build  && terraform plan -var "cluster={zone=\"$(GCP_CLUSTER_ZONE)\"}" \
+	-var "cluster={machineType=\"$(GCP_CLUSTER_NODEPOOL_MACHINETYPE)\"}" -var "cluster={name=\"$(GCP_CLUSTER_NAME)\"}" \
+	-var "cluster={project=\"$(GCP_PROJECT)\"}" \
+	-var "cluster={password=\"$(GCP_PASSWORD)\"}" \
+	-var "cluster={initialNodeCount=\"$(GCP_CLUSTER_NODEPOOL_INITIALNODECOUNT)\"}" -var "cluster={legacyABAC=\"$(GCP_CLUSTER_LEGACYABAC)\"}" -out tfplan'
+	docker run --rm -it $(common_mounts) $(DOCKER_RUN_ARGS) $(build_tag) bash -c ' \
+	cd $(mount_path)/build && terraform apply -input=false tfplan'
```

Comment:
Curious - tfplan contains all previous variable defined ?

Are we ok having that file with our password left behind ? if you can mix tfplan file and env var I'd like the password to be pass after at the apply time so nothing is left behind.

Reply:
I am currently investigating what we can do, all variables are stored inside tfplan so on the apply step we can omit them.